requirements/build.txt

```diff
@@ -1,12 +1 @@
-# Generated via tools/generate_requirements.py and pre-commit hook.
-# Do not edit this file; modify pyproject.toml instead.
-meson-python>=0.15
-wheel
-setuptools>=67
-packaging>=21
-ninja
-Cython>=3.0.4
-pythran
-numpy>=2.0.0rc1
-spin==0.8
-build
+.[build]
```

Comment:
Maybe I'm missing something, but I don't think this is an equivalent replacement. If I run `pip install -r requirements/build.txt` it installs scikit-image as well where previously we could setup the build environment without having to install scikit-image as well. Not sure if we want or need to keep this ability.

If not, then why not invoke `pip install .[build]` directly and forgo requirement files alltogether?

Reply:
Err, you are right; this is not what we want.

The alternative approach (where everything is defined in `pyproject.toml`) is [install-deps](https://gist.github.com/stefanv/0b052fa4014fa07e18e81fe544afc9f9).

Reply:
> If not, then why not invoke `pip install .[build]` directly and forgo requirement files alltogether?

That what I would have done in the next step, but I don't know if external services (readthedocs) have configuration that point to those files, so I didn't wanted to remove them.